chain/bitcoind_conn.go

```diff
@@ -311,9 +327,55 @@ func (c *BitcoindConn) sendTxToClients() {
 	}
 }
 
+// getBlockHashDuringStartup is used to call the getblockhash RPC during
+// startup. It catches the case where bitcoind is still in the process of
+// loading blocks, which returns the following error,
+// - "-28: Loading block index..."
+// - "-28: Verifying blocks..."
+// In this case we'd retry every second until we time out after 30 seconds.
+func getBlockHashDuringStartup(
+	client *rpcclient.Client) (*chainhash.Hash, error) {
+
+	hash, err := client.GetBlockHash(0)
+
+	// Exit early if there's no error.
+	if err == nil {
+		return hash, nil
+	}
+
+	// If the error doesn't start with "-28", it's an unexpected error so
+	// we exit with it.
+	if !strings.Contains(err.Error(), errStillLoadingCode) {
+		return nil, err
+	}
+
+	// Otherwise, we'd retry calling getblockhash or time out after 30s.
+	for {
+		select {
+		// Timeout after 30s.
+		case <-time.After(bitcoindStartTimeout):
```

Comment:
Sorry, I missed this before. Because we create a new channel with `time.After()` within the select, this starts ticking afresh for each loop we make. So the one second one will always and forever tick first and this one will never get to tick. We need to create the channel outside of the loop if we want to cancel re-trying after 30 seconds.

Reply:
oh shoot my bad, now fixed!